src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -280,15 +287,103 @@ private boolean parseNodes(List<Node> nodes) {
           continue;
         }
 
-        if (cpeUriCheck(cpeMatch)) {
+        if (projectCheck(cpeMatch, references)) {
           return true;
         }
       }
 
-      if (parseNodes(node.getChildren())) {
+      if (parseNodes(node.getChildren(), references)) {
         return true;
       }
     }
     return false;
   }
+
+  /**
+   * Checks if the input {@link URI} is same as the project's URL.
+   * 
+   * @param referenceUrl of type {@link URI}.
+   * @return True if the input URL matches the project's URL. Otherwise false.
+   */
+  private boolean checkUrl(URI referenceUrl) {
+    if (referenceUrl.getHost().equals(project.url().getHost())
+        && referenceUrl.getScheme().equals(project.url().getProtocol())
+        && checkProjectUrl(referenceUrl.getPath())) {
+      return true;
+    }
+    return false;
```

Comment:
Would you mind simplifying this?

```suggestion
    return referenceUrl.getHost().equals(project.url().getHost())
        && referenceUrl.getScheme().equals(project.url().getProtocol())
        && checkProjectUrl(referenceUrl.getPath());
```

Does it really need checking schemas?